Gems/Atom/RPI/Code/Source/RPI.Reflect/Model/ModelAsset.cpp

```diff
@@ -248,8 +236,15 @@ namespace AZ
                 AZ::Vector3 intersectionNormal;
 
                 float shortestDistanceNormalized = AZStd::numeric_limits<float>::max();
-                const AZ::u32* indexPtr = reinterpret_cast<const AZ::u32*>(indexRawBuffer.data());
-                for (uint32_t indexIter = 0; indexIter <= indexRawDesc.m_elementCount - 3; indexIter += 3, indexPtr += 3)
+
+                const AZ::u32* indexPtr = reinterpret_cast<const AZ::u32*>(
+                    indexRawBuffer.data() + (indexBufferViewDesc.m_elementOffset * indexBufferViewDesc.m_elementSize));
+                const float* positionPtr = reinterpret_cast<const float*>(
+                    positionRawBuffer.data() + (positionBufferViewDesc.m_elementOffset * positionBufferViewDesc.m_elementSize));
+
+                constexpr int StepSize = 3; // number of values per vertex (x, y, z)
+                for (uint32_t indexIter = 0; indexIter <= indexBufferViewDesc.m_elementCount - StepSize;
```

Comment:
Wouldn't this be equivalent to using < rather than <= and losing the - StepSize? Seems like that would be slightly easier to read.

Reply:
That's true, this preceded my change but I can update it, good call :+1: